components/fxa-client/src/internal/push.rs

```diff
@@ -83,6 +80,69 @@ impl FirefoxAccount {
             }
         }
     }
+
+    /// Process a push message, updating the internal state and emit FxaEvent
+    pub fn process_push_message(&mut self, payload: &str) -> Result<()> {
```

Comment:
I did my best to code this correctly, but another set of eyes or two would be great.